File: core-protocol/src/test/kotlin/in/meshsetu/protocol/SchedulerTest.kt
package `in`.meshsetu.protocol

import `in`.meshsetu.model.EncryptedObject
import `in`.meshsetu.model.TrafficClass
import kotlin.test.Test
import kotlin.test.assertEquals

class SchedulerTest {
    @Test fun emergencyPreemptsBulkAndExpiryIsDropped() {
        val q = OutboundScheduler()
        q.enqueue(EncryptedObject(1u, TrafficClass.VOICE_EVIDENCE, byteArrayOf(1), 100), 0)
        q.enqueue(EncryptedObject(2u, TrafficClass.SOS_STRUCTURED, byteArrayOf(1), 100), 0)
        q.enqueue(EncryptedObject(3u, TrafficClass.ROOM_MESSAGE, byteArrayOf(1), 1), 0)
        assertEquals(2u, q.next(0)?.objectId)
        assertEquals(1u, q.next(2)?.objectId)
    }
}

File: mobile/test/core/protocol/scheduler_test.dart
import 'dart:typed_data';

import 'package:meshsetu_mobile/core/model/model.dart';
import 'package:meshsetu_mobile/core/protocol/outbound_scheduler.dart';
import 'package:test/test.dart';

void main() {
  test('emergency preempts bulk and expiry is dropped', () {
    final q = OutboundScheduler();
    q.enqueue(
      EncryptedObject(
        objectId: 1,
        trafficClass: TrafficClass.voiceEvidence,
        bytes: Uint8List.fromList([1]),
        expiresAtMs: 100,
      ),
      0,
    );
    q.enqueue(
      EncryptedObject(
        objectId: 2,
        trafficClass: TrafficClass.sosStructured,
        bytes: Uint8List.fromList([1]),
        expiresAtMs: 100,
      ),
      0,
    );
    q.enqueue(
      EncryptedObject(
        objectId: 3,
        trafficClass: TrafficClass.roomMessage,
        bytes: Uint8List.fromList([1]),
        expiresAtMs: 1,
      ),
      0,
    );
    expect(q.next(0)?.objectId, 2);
    expect(q.next(2)?.objectId, 1);
  });
}
